hbase-client/src/main/java/org/apache/hadoop/hbase/client/AsyncRegionLocator.java

```diff
@@ -85,26 +118,41 @@ private boolean isMeta(TableName tableName) {
     return TableName.isMetaTableName(tableName);
   }
 
+  private AbstractAsyncTableRegionLocator getOrCreateTableRegionLocator(TableName tableName) {
```

Comment:
Can't return an Interface rather than Abstract class?

Reply:
This is internal so just use the implementation class directly. Can have an interface later if we think it is useful.